File: src/main/java/tickets/Ticket.java
package tickets;

import flights.Flight;

public class Ticket {
    private Passenger passenger;
    private Flight flight;
    private int ticketNum;
    private TicketType ticketType;
    private TicketPool ticket;


    public Ticket(Passenger p, int ticketNum) {
        this.passenger = p;
        this.ticketNum = ticketNum;
    }

    public Passenger getPassenger() {
        return passenger;
    }

    public void setPassenger(Passenger passenger) {
        this.passenger = passenger;
    }

    public Flight getFlight() {
        return flight;
    }

    public void setFlight(Flight flight){
        this.flight = flight;
    }

    public int getTicketNum() {
        return ticketNum;
    }

    public void setTicketNum(int ticketNum) {
        this.ticketNum = ticketNum;
    }

    public TicketPool getTicket() {
        return ticket;
    }

}

File: src/main/java/tickets/Ticket.kt
package tickets


import flights.Flight

/**
 * Ticket class. Contains the information about the ticket.
 */
open class Ticket {
    var passenger: Passenger? = null
    lateinit var flight: Flight
    var ticketNum: String
    var ticketType: TicketType = TicketType.TICKET

    //@param ticketNum, flight   when flight is known
    constructor(ticketNum: String, flight: Flight) {
        this.ticketNum = ticketNum
        this.flight = flight
    }

    //@param p, ticketNum   when flight is not known
    constructor(p: Passenger?, ticketNum: String) {
        passenger = p
        this.ticketNum = ticketNum
    }
}